src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -443,11 +448,24 @@ export async function deleteDataSourceById(
 
 export async function deleteMultipleDataSources(
   savedObjectsClient: SavedObjectsClientContract,
-  selectedDataSources: DataSourceTableItem[]
+  selectedDataSources: DataSourceTableItem[],
+  http: HttpSetup
 ) {
   await Promise.all(
     selectedDataSources.map(async (selectedDataSource) => {
-      await deleteDataSourceById(selectedDataSource.id, savedObjectsClient);
+      if (selectedDataSource.objectType === DATA_CONNECTION_SAVED_OBJECT_TYPE) {
+        const savedObject = await savedObjectsClient.get(
+          DATA_CONNECTION_SAVED_OBJECT_TYPE,
+          selectedDataSource.id
+        );
+        const dataSourceMDSId = savedObject.references.find((r) => r.type === 'data-source')?.id;
+        return http.delete(
+          `${DATACONNECTIONS_BASE}/${selectedDataSource.title}/dataSourceMDSId=${
+            dataSourceMDSId || ''
+          }`
+        );
+      }
+      return deleteDataSourceById(selectedDataSource.id, savedObjectsClient);
```

Comment:
Do we need to handle scenarios where one deletion fails either saved object or SQL and mention it to the user explicitly?

Reply:
Some CTA for users as toast to understand what's the inconsistent state.

Reply:
That should be existing logic when calling this utils. The utils only deletes and throws error up https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f129a7e48194aba1cb059c67a68a817a73a4139d/src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx#L451-L474